Source/Scene/ColorConditional.js

```diff
@@ -0,0 +1,130 @@
+/*global define*/
+define([
+       './Expression',
+       '../Core/clone',
+       '../Core/Color',
+       '../Core/defaultValue',
+       '../Core/defined',
+       '../Core/defineProperties'
+    ], function(
+        Expression,
+        clone,
+        Color,
+        defaultValue,
+        defined,
+        defineProperties) {
+    'use strict';
+
+    /**
+     * DOC_TBA
+     * <p>
+     * Do not construct this directly; instead use {@link Cesium3DTileStyle}.
+     * </p>
+     */
+    function ColorConditional(styleEngine, jsonExpression) {
+        this._styleEngine = styleEngine;
+        this._conditional = clone(jsonExpression.conditional, true);
+        this._default = jsonExpression.default;
+
+        this._runtimeConditional = undefined;
+        this._runtimeDefault = undefined;
+
+        setRuntime(this);
+        setRuntimeDefault(this);
+    }
+
+    defineProperties(ColorConditional.prototype, {
```

Comment:
I would actually remove this for now until we expose the AST (probably not for the first pull request into `3d-tiles`).

Otherwise, this looks like a fast get/set, but it is just as bad as assigning a new style since it forces a recompile.